### ISSUE TITLE
Forms: User with AT feedback

### ISSUE THREAD

**Issue by bsouster** [wet-boew/wet-boew#8916]
User testing with [Steps form page](http://universallabs.org/wet/labs/pr-8882/demos/stepsform/stepsform-en.html):

Tools: 
*Keyboard-only navigation
*Speech recognition software (Dragon Naturally Speaking)

General:
*In IE11, the required date formatting is lost with the missing field placeholder. Date format needs to be shown in the field label as well... this will also help users who need static formatting information for reference after placeholders are hidden during input activity.

Keyboard-only navigation:
*In Firefox, the initial focus highlight (before a selection is made) is very difficult to see. Technically, it passes WCAG 2.1. In practice, it is not accessible.

Speech recognition:
*No issues noted specific to the experience with speech recognition software

**Comment by GormFrank** [comment]
### IE11
A placeholder could be added to the working example page to fix that problem.

### Firefox
WET lets browsers show their own native date picker if they have one. Firefox has one. If there is an accessibility issue with a native date picker, then technically it is the browser's fault since browsers should be accessible by default (at the very least).

**Comment by EricDunsworth** [comment]
> **General issues**
> In IE11, the required date formatting is lost with the missing field placeholder. Date format needs to be shown in the field label as well... this will also help users who need static formatting information for reference after placeholders are hidden during input activity.
>> IE11
>> A placeholder could be added to the working example page to fix that problem.

I don't think there'd be any need for a ``placeholder`` attribute on the date field.

The main issue here seems to be that ``<span class="datepicker-format">...</span>`` (as seen in the [date picker polyfill demo](https://wet-boew.github.io/v4.0-ci/docs/ref/datepicker/datepicker-en.html)) is missing from the end of the "Date of Birth" label in the [steps forms demo](https://wet-boew.github.io/v4.0-ci/demos/stepsform/stepsform-en.html).

In browsers that lack native support for ``<input type="date">`` (like IE11), ``<span class="datepicker-format">...</span>`` is shown in the labels of date inputs. In browsers that have native support (i.e. just about everything else), the YYYY-MM-DD format is usually shown inside date fields and can be changed interactively via a keyboard. As a result, ``<span class="datepicker-format">...</span>`` is hidden in modern browsers to reduce bloat.

> **Using keyboard-only navigation**
> In Firefox, the initial focus highlight (before a selection is made) is very difficult to see. Technically, it passes WCAG 2.1. In practice, it is not accessible. It will fail WCAG 2.2 if the [Focus Visible (enhanced) (AA)](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible-enhanced.html) reaches release.
>> **Firefox**
>> WET lets browsers show their own native date picker if they have one. Firefox has one. If there is an accessibility issue with a native date picker, then technically it is the browser's fault since browsers should be accessible by default (at the very least).

@GormFrank I don't think @bsouster was talking about the date picker in that part. I think he meant the focus indicator on form controls. It uses a light blue glowing effect on normal fields and a red glowing effect on fields errors have been detected in. That effect comes from WET's CSS (via Bootstrap 3.x).  That effect is probably used in pretty much all WET's forms.

**Comment by thekodester** [comment]
Just an FYI for Firefox. It does have a lot of accessibility issues right now. Last time I check the bug reports (https://bugzilla.mozilla.org/home) MacOS voiceover was completely broken for example.
